gpkitmodels/Commercial Sizing/CFP_Fuselage_Performance.py

```diff
@@ -0,0 +1,978 @@
+
+        with vectorize(Ncruise):
+            crs = CruiseSegment(ac)
+
+        #declare new variables
+        W_ftotal = Variable('W_{f_{total}}', 'N', 'Total Fuel Weight')
+        W_fclimb = Variable('W_{f_{climb}}', 'N', 'Fuel Weight Burned in Climb')
+        W_fcruise = Variable('W_{f_{cruise}}', 'N', 'Fuel Weight Burned in Cruise')
+        W_total = Variable('W_{total}', 'N', 'Total Aircraft Weight')
+        CruiseAlt = Variable('CruiseAlt', 'ft', 'Cruise Altitude [feet]')
+        ReqRng = Variable('ReqRng', 'nautical_miles', 'Required Cruise Range')
+
+        h = cls.state['h']
+        hftClimb = cls.state['hft']
+        dhft = cls.climbP['dhft']
+        hftCruise = crs.state['hft']
+
+        #make overall constraints
+        constraints = []
+
+        constraints.extend([
+            #weight constraints
+            TCS([ac['W_{fuse}'] + ac['W_{pay}'] + W_ftotal + ac['numeng'] * ac['W_{engine}'] + ac['W_{wing}'] <= W_total]),
+
+            cls.climbP.aircraftP['W_{start}'][0] == W_total,
+            cls.climbP.aircraftP['W_{end}'][-1] == crs.cruiseP.aircraftP['W_{start}'][0],
+
+            # similar constraint 1
+            TCS([cls.climbP.aircraftP['W_{start}'] >= cls.climbP.aircraftP['W_{end}'] + cls.climbP.aircraftP['W_{burn}']]),
+            # similar constraint 2
+            TCS([crs.cruiseP.aircraftP['W_{start}'] >= crs.cruiseP.aircraftP['W_{end}'] + crs.cruiseP.aircraftP['W_{burn}']]),
+
+            cls.climbP.aircraftP['W_{start}'][1:] == cls.climbP.aircraftP['W_{end}'][:-1],
+            crs.cruiseP.aircraftP['W_{start}'][1:] == crs.cruiseP.aircraftP['W_{end}'][:-1],
+
+            TCS([ac['W_{fuse}'] + ac['W_{pay}'] + ac['numeng'] * ac['W_{engine}'] + ac['W_{wing}'] <= crs.cruiseP.aircraftP['W_{end}'][-1]]),
+
+            TCS([W_ftotal >=  W_fclimb + W_fcruise]),
+            TCS([W_fclimb >= sum(cls.climbP['W_{burn}'])]),
+            TCS([W_fcruise >= sum(crs.cruiseP['W_{burn}'])]),
+
+            #altitude constraints
+            hftCruise == CruiseAlt,
+            TCS([hftClimb[1:Ncruise] >= hftClimb[:Ncruise-1] + dhft]),
+            TCS([hftClimb[0] >= dhft[0]]),
+            hftClimb[-1] <= hftCruise,
+
+            #compute the dh
+            dhft == hftCruise/Nclimb,
+
+            #constrain the thrust
+            cls.climbP.engineP['thrust'] <= 2 * max(crs.cruiseP.engineP['thrust']),
+
+            #set the range for each cruise segment, doesn't take credit for climb
+            #down range disatnce covered
+            crs.cruiseP['Rng'] == ReqRng/(Ncruise),
+
+            #set the TSFC
+            cls.climbP.engineP['TSFC'] == .7*units('1/hr'),
+            crs.cruiseP.engineP['TSFC'] == .5*units('1/hr'),
+            ])
+        
+        # Model.__init__(self, W_ftotal + s*units('N'), constraints + ac + cls + crs, subs)
+        Model.__init__(self, W_ftotal, constraints + ac + cls + crs, subs)
+
+    def bound_all_variables(self, model, eps=1e-30, lower=None, upper=None):
```

Comment:
this junk DNA keeps floating around...the version of this in master (in `constraints.bounded`) should be superior.